lib/fw.c

```diff
@@ -970,9 +962,20 @@ static int switchtec_fw_part_info(struct switchtec_dev *dev, int nr_info,
 	int ret;
 	int i;
 
+	uint8_t subcmd = MRPC_PART_INFO_GET_ALL_INFO;
+	struct switchtec_flash_info_gen4 all_info;
+
 	if (info == NULL || nr_info == 0)
 		return -EINVAL;
 
+	if (dev->gen == SWITCHTEC_GEN4) {
+		ret = switchtec_cmd(dev, MRPC_PART_INFO, &subcmd,
+				    sizeof(subcmd), &all_info,
+				    sizeof(all_info));
+		if (ret < 0)
```

Comment:
Any non-zero value should be taken as error.

Reply:
Updated, please review.